transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -1053,4 +1079,53 @@ boolean setClosed() {
             return super.trySuccess();
         }
     }
+
+    private static final class AutoFlushTask implements Runnable {
+
+        private final AbstractChannel channel;
+        private final SingleThreadEventLoop eventLoop;
+        private boolean writePending;
+        private boolean running;
+
+        AutoFlushTask(AbstractChannel channel) {
+            this.channel = channel;
+            assert channel.eventLoop() instanceof SingleThreadEventLoop;
+            eventLoop = (SingleThreadEventLoop) channel.eventLoop();
+        }
+
+        @Override
+        public void run() {
+            running = true;
+            try {
+                if (writePending) {
+                    writePending = false;
+                    channel.flush();
+                    if (writePending) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)